src/cpu/Dasm.cc

```
@@ -23,82 +22,90 @@ void appendAddrAsHex(std::string& output, uint16_t addr)
 	strAppend(output, '#', hex_string<4>(addr));
 }
 
-unsigned dasm(const MSXCPUInterface& interface, uint16_t pc, std::span<uint8_t, 4> buf,
-              std::string& dest, EmuTime::param time,
+constexpr std::array<uint8_t, 28> ddcb__ = {
+	// keep'em in order
+	0x06, 0x0e, 0x16, 0x1e, 0x46, 0x4e, 0x56, 0x5e, 0x66, 0x6e, 0x76, 0x7e, 0x86, 0x8e,
+	0x96, 0x9e, 0xa6, 0xae, 0xb6, 0xbe, 0xc6, 0xce, 0xd6, 0xde, 0xe6, 0xee, 0xf6, 0xfe
+};
+
+std::optional<unsigned> instructionLength(std::span<const uint8_t> bin)
+{
+	if (bin.empty()) return {};
+	auto t = instr_len_tab[bin[0]];
+	if (t < 4) return t;
+
+	if (bin.size() < 2) return 1;
```

I don't agree with this change. Let's take an example: `ED 44 (neg)`, this is a valid instruction.
But if we only pass `ED` then this function now returns `1`. Instead it should report an error (via `std::nullopt`), meaning "insufficient data to calculate the instruction length".